### PR TITLE
Greedy Algorithm Implementations and Explanations #4493

### PR DESCRIPTION
**Pull request by Vineetttt**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please remove `main`, logging to console, and add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by Vineetttt** [comment]
> The code looks good, could you please remove `main`, logging to console, and add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

@siriak Yes I'll do the changes but can you elaborate a bit, what exactly needs to be done

**Comment by Vineetttt** [comment]
@siriak I've made the necessary changes you can check.

**Comment by Vineetttt** [comment]
@siriak I've made the necessary changes you can check and merge accordingly.

**Review by Vineetttt — COMMENTED** [review]
I've made the necessary changes you can check and merge accordingly.

File: src/main/java/com/thealgorithms/greedyalgorithms/ActivitySelection.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        // Iterate through sorted activities to select compatible ones.
+        for (int i = 1; i < n; i++) {
+            if (activities[i][1] >= lastEnd) {                
+                ans.add(activities[i][0]); 
+                lastEnd = activities[i][2]; 
+            }
+        }
+        return ans;
+    }
+    public static void printActivities(ArrayList<Integer> ans) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/greedyalgorithms/ActivitySelection.java:40]: No need to print anything, presentation should be separate from the algorithm

File: src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+public class CoinChangeTest {
+    @Test
+    public void testCoinChangeProblemWithValidAmount() {
+        ArrayList<Integer> expected = new ArrayList<>(Arrays.asList(500, 50, 20, 20, 1));
```
Comment (siriak) [inline src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java:12]: Please add tests where not all coins are used.

File: src/test/java/com/thealgorithms/greedyalgorithms/DijkstrasAlgorithmTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.thealgorithms.greedyalgorithms;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class DijkstrasAlgorithmTest {
+    @Test
+    public void testDijkstraShortestPath() {
+        int[][] graph = {{0, 2, 4, 0, 0, 0}, {0, 0, 1, 7, 0, 0}, {0, 0, 0, 0, 3, 0}, {0, 0, 0, 0, 0, 1}, {0, 0, 0, 2, 0, 5}, {0, 0, 0, 0, 0, 0}};
```
Comment (siriak) [inline src/test/java/com/thealgorithms/greedyalgorithms/DijkstrasAlgorithmTest.java:10]: Dijkstras algorithm is already implemented

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix PR checks

**Review by Vineetttt — COMMENTED** [review]
Appropriate changes are made, can you check.

File: src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java (hunk 2 above)
Comment (siriak) [reply src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java:12]: This is not addressed

**Comment by siriak** [comment]
Please fix the linter job which is failing and add more tests for coin change

**Comment by Vineetttt** [comment]
@siriak Necessary changes are made, you can check .

File: src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java (hunk 2 above)
Comment (siriak) [reply src/test/java/com/thealgorithms/greedyalgorithms/CoinChangeTest.java:12]: This is not addressed

**Review by siriak — CHANGES_REQUESTED** [review]

**Review by siriak — APPROVED** [review]

**Pull request merged**